test/e2e.test.ts

```diff
@@ -0,0 +1,125 @@
+import { ChildProcessWithoutNullStreams, execSync, spawn } from 'child_process';
+import { chmodSync } from 'fs';
+import chalk from 'chalk';
+import { createTmpDir } from './utils';
+import { getAvailableTemplates } from '../src/cli/cli';
+
+const CLI_EXECUTABLE = `${__dirname}/../dist/cli/index.js`;
+// Turning this flag to 'true' will print each command before executing it
+// It might be useful to turn on, while debugging particular test.
+const DEBUG_COMMANDS = false;
+
+it('has disabled DEBUG_COMMANDS flag', () => {
+  expect(DEBUG_COMMANDS).toBe(false);
+});
+
+type CommandArg = [string, string | string[] | number | boolean];
+const execCommand = (command: string, ...args: CommandArg[]) => {
+  const quote = (val: string) => `"${val}"`;
+  const formattedArgs = args
+    .map(([c, a]) => {
+      // if args is array then quote each elem and separate them with space
+      if (Array.isArray(a)) return `${c} ${a.map(quote).join(' ')}`;
+      // otherwise just quote each elem and separate them with space
+      else return `${c} ${quote(String(a))}`;
+    })
+    .join(' ');
+  const formattedCommand = `${command} ${formattedArgs}`;
+  if (DEBUG_COMMANDS) console.log(`Executing command: ${formattedCommand}`);
+  try {
+    return execSync(`${CLI_EXECUTABLE} ${formattedCommand}`).toString().trim();
+  } catch (e: any) {
+    // rethrow the output of the CLI
+    throw new Error(e.stdout.toString().trim());
+  }
+};
+
+describe('Beacon reader CLI', () => {
+  beforeAll(() => {
+    chmodSync(CLI_EXECUTABLE, '755');
+  });
+
+  it('shows help', () => {
+    const output = execCommand('--help');
+    expect(output).toMatchSnapshot();
+  });
+
+  it('shows available commands', () => {
+    const output = execCommand('show-available-templates');
+    expect(output).toEqual(['javascript-ethers'].join('\n'));
+  });
+
+  describe('interactive command', () => {
+    const waitOnOutput = (childProcess: ChildProcessWithoutNullStreams) =>
+      new Promise<string>((resolve) =>
+        childProcess.stdout.once('readable', () => {
+          const data = childProcess.stdout.read().toString();
+          resolve(data);
+        })
+      );
+
+    it('works', async () => {
```

Comment:
I am not entirely happy how this test looks like, but this was the best I could do to make it work.

Reply:
Actually I needed to add one more setTimeout after the path is written to stdin of child process. THat made that test somewhat reliable.